commit 15241406018b861ea4441786cb5f424c9e308034
Author: S.Çağlar Onur <caglar@cs.princeton.edu>
Date:   Thu Nov 18 16:25:37 2010 -0500

    ACPI: Idle C-states disabled by max_cstate should not disable the TSC

diff --git a/drivers/acpi/processor_idle.c b/drivers/acpi/processor_idle.c
index 94483a2..5812380 100644
--- a/drivers/acpi/processor_idle.c
+++ b/drivers/acpi/processor_idle.c
@@ -1158,7 +1158,7 @@ static int acpi_processor_power_verify(struct acpi_processor *pr)
 
 	pr->power.timer_broadcast_on_state = INT_MAX;
 
-	for (i = 1; i < ACPI_PROCESSOR_MAX_POWER; i++) {
+	for (i = 1; i < ACPI_PROCESSOR_MAX_POWER && i <= max_cstate; i++) {
 		struct acpi_processor_cx *cx = &pr->power.states[i];
 
 		switch (cx->type) {
